Mathlib/Analysis/SpecialFunctions/MulExpNegSq.lean

```diff
@@ -0,0 +1,187 @@
+  rw [mulExpNegSq_apply, neg_mul, ← le_div_iff₀ (Real.exp_pos (-(x * x))), Real.exp_neg,
+    div_inv_eq_mul, one_mul]
+  apply le_trans _ (Real.add_one_le_exp (x * x))
+  nlinarith
+
+theorem mulExpNegSq_bounded (x : ℝ) : abs (mulExpNegSq x) ≤ 1 := by
+  apply abs_le.mpr ⟨_, mulExpNegSq_bounded_above x⟩
+  rw [← mulExpNegSq_symm, neg_le_neg_iff]
+  exact mulExpNegSq_bounded_above (-x)
+
+theorem expNegSq_deriv (y : ℝ) :
+    deriv (fun (x : ℝ) => (-x * x).exp) y = -2 * y * (-y * y).exp := by
+  simp only [neg_mul, differentiableAt_neg_iff, differentiableAt_id', DifferentiableAt.mul,
+    deriv_exp, deriv.neg', deriv_mul, deriv_id'', one_mul, mul_one, neg_add_rev]
+  ring
+
+theorem expNegSq_differentiableAt (y : ℝ) :
+    DifferentiableAt ℝ (fun (x : ℝ) => (-x * x).exp) y := by
+  simp only [neg_mul, differentiableAt_neg_iff, differentiableAt_id', DifferentiableAt.mul,
+    DifferentiableAt.exp, implies_true]
+
+theorem expNegSq_differentiable : Differentiable ℝ (fun (x : ℝ) => (-x * x).exp) :=
+  fun _ => expNegSq_differentiableAt _
+
+theorem mulExpNegSq_deriv (x : ℝ) : deriv (fun x => x * (Real.exp (- x * x))) x =
+    (-x * x).exp + x * (-2 * x * (-x * x).exp) := by
+  nth_rw 1 [← one_mul (-x * x).exp, ← (deriv_id x)]
+  rw [← expNegSq_deriv x]
+  apply deriv_mul differentiableAt_id' (expNegSq_differentiableAt x)
+
+theorem mulExpNegSq_deriv_le_one (x : ℝ) : ‖deriv (fun y ↦ y * (-y * y).exp) x‖₊ ≤ 1 := by
+  rw [← NNReal.coe_le_coe, coe_nnnorm, Real.norm_eq_abs, mulExpNegSq_deriv x]
+  have heq : (-x * x).exp + x * (-2 * x * (-x * x).exp) = (-x * x).exp * (1 + 2 * -x * x) := by
+    ring
+  rw [heq, abs_mul, Real.abs_exp]
+  let y := x * x
+  have hy : y = x * x := by rfl
+  rw [neg_mul, mul_assoc, neg_mul, ← hy]
+  apply mul_le_of_le_inv_mul₀ (zero_le_one' ℝ) (Real.exp_nonneg _)
+  simp [← Real.exp_neg (-y), abs_le]
+  constructor
+  · have twomul_le_exp : 2 * y ≤ y.exp := by
+      apply le_trans _ Real.emul_le_exp
+      have : 2 ≤ Real.exp 1 := by
+        apply le_of_lt (lt_trans _ Real.exp_one_gt_d9)
+        norm_num
+      apply mul_le_mul_of_nonneg_right this (mul_self_nonneg x)
+    apply le_trans twomul_le_exp _
+    simp only [le_add_iff_nonneg_right, zero_le_one]
+  · apply le_trans (Real.one_le_exp (mul_self_nonneg x)) (le_add_of_nonneg_right _)
+    simp [hy, (mul_self_nonneg x)]
+
+theorem mulExpNegSq_lipschitzWith_one : LipschitzWith 1 mulExpNegSq := by
+  apply lipschitzWith_of_nnnorm_deriv_le
+      (Differentiable.mul differentiable_id' expNegSq_differentiable) mulExpNegSq_deriv_le_one
+
+end mulExpNegSq
+
+section mulExpNegMulSq
+
+/-! ### Definition and properties of `mulExpNegMulSq g ε` -/
+
+variable {E: Type*}
+
+/--
+`mulExpNegMulSq` transforms a continuous function `g` into another continuous function with useful
+boundedness and convergence properties.
+-/
+noncomputable
+def mulExpNegMulSq [TopologicalSpace E] {g : E → ℝ} (hg: Continuous g) (ε : ℝ) :
+    ContinuousMap E ℝ where
```

Comment:
It's a little strange that you take a function and continuity separately on the input side, but bundle them on the output side. Unless there's a good reason I'm missing, probably they should just be separate? You can add a `@[continuity]` attribute to the theorem.